alfresco-transformer-base/src/test/java/org/alfresco/transformer/AbstractTransformerControllerTest.java

```diff
@@ -287,4 +299,134 @@ public void testEmptyPojoTransform() throws Exception
         // Assert the reply
         assertEquals(BAD_REQUEST.value(), transformReply.getStatus());
     }
+
+    @Test
+    public void testGetTransformConfigInfo() throws Exception
+    {
+        TransformConfig expectedTransformConfig = objectMapper
+            .readValue(new ClassPathResource("engine_config.json").getFile(),
+                TransformConfig.class);
+
+        ReflectionTestUtils
+            .setField(AbstractTransformerController.class, "ENGINE_CONFIG", "engine_config.json");
+
+        String response = mockMvc.perform(MockMvcRequestBuilders.get("/info"))
+            .andExpect(status().is(OK.value())).andExpect(
+                header().string(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_UTF8_VALUE))
+            .andReturn().getResponse().getContentAsString();
+
+        TransformConfig transformConfig = objectMapper.readValue(response, TransformConfig.class);
+
+        assertEquals(expectedTransformConfig, transformConfig);
+    }
+
+    @Test
+    public void testGetInfoFromConfigWithDuplicates() throws Exception
+    {
+        TransformConfig expectedResult = buildCompleteTransformConfig();
+
+        ReflectionTestUtils.setField(AbstractTransformerController.class, "ENGINE_CONFIG",
+            "engine_config_with_duplicates.json");
+
+        String response = mockMvc.perform(MockMvcRequestBuilders.get("/info"))
+            .andExpect(status().is(OK.value())).andExpect(
+                header().string(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_UTF8_VALUE))
+            .andReturn().getResponse().getContentAsString();
+
+        TransformConfig transformConfig = objectMapper.readValue(response, TransformConfig.class);
+
+        assertNotNull(transformConfig);
+        assertEquals(expectedResult, transformConfig);
+        assertEquals(3, transformConfig.getTransformOptions().get("engineXOptions").size());
+        assertEquals(1,
+            transformConfig.getTransformers().get(0).getSupportedSourceAndTargetList().size());
+        assertEquals(1,
+            transformConfig.getTransformers().get(0).getTransformOptions().size());
+
+    }
+
+    @Test
+    public void testGetInfoFromIncompleteConfig() throws Exception
```

Comment:
Add comment to explain what "incomplete" means ?

Again, if a negative test shouldn't it result in a (startup) error ?

Reply:
Some TEngines can have no `transformOptions` (e.g. _LibreOffice_); I think this is the case with this particular test. Maybe we should rename the test.